psm/src/arch/wasm32.s

```diff
@@ -1,25 +1,48 @@
 #include "psm.h"
 
-.text
-.section .text.rust_psm_stack_direction,"",@
+# Note that this function is not compiled when this package is uploaded to
+# crates.io, this source is only here as a reference for how the corresponding
+# wasm32.o was generated. This file can be compiled with:
+#
+#    cpp psm/src/arch/wasm32.s | llvm-mc -o psm/src/arch/wasm32.o --arch=wasm32 -filetype=obj
+#
+# where you'll want to ensure that `llvm-mc` is from a relatively recent
+# version of LLVM.
+
+.globaltype __stack_pointer, i32
+
 .globl rust_psm_stack_direction
 .type rust_psm_stack_direction,@function
 rust_psm_stack_direction:
 .functype rust_psm_stack_direction () -> (i32)
-    i32.const $STACK_DIRECTION_DESCENDING
+    i32.const STACK_DIRECTION_DESCENDING
     end_function
-.rust_psm_stack_direction_end:
-.size rust_psm_stack_direction, .rust_psm_stack_direction_end-rust_psm_stack_direction
 
-
-.section .text.rust_psm_stack_pointer,"",@
 .globl rust_psm_stack_pointer
 .type rust_psm_stack_pointer,@function
 rust_psm_stack_pointer:
 .functype rust_psm_stack_pointer () -> (i32)
     global.get __stack_pointer
     end_function
-.rust_psm_stack_pointer_end:
-.size rust_psm_stack_pointer, .rust_psm_stack_pointer_end-rust_psm_stack_pointer
 
-.globaltype __stack_pointer, i32
+.globl rust_psm_on_stack
+.type rust_psm_on_stack,@function
```

Comment:
You may need to add another similar implementation for `rust_psm_replace_stack`. If the crate(s) work on wasm right now, its probably only because stacker or its test suite itself doesn't use the function, but `psm`'s public API does depend on it.

Reply:
Sure thing, added!